### PR TITLE
Add new command 'update:check' in symfony console

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Add new command 'update:check' in symfony console, allowing you to you to check the different prerequisites for an update
| Type?             new feature
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | -
| Sponsor company   | -
| How to test?      | See README.md for new command

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
* Please add the command details in the readme

File: README.md [hunk 1]
```diff
@@ -63,6 +63,12 @@ To use the Symfony Console, simply run the following command from the root direc
 $ php bin/console
 ```
 
+A command allowing you to check the different prerequisites for an update is available
```
Comment (Quetzacoalt91) [inline README.md]:
```suggestion
The requirements can be reviewed to confirm the shop is safe to update:
```

File: README.md [hunk 2]
```diff
@@ -63,6 +63,12 @@ To use the Symfony Console, simply run the following command from the root direc
...
+
+```
+$ php bin/console update:check <your-admin-dir>
+```
+
 For instance, to start upgrading your store, you would use:
```
Comment (Quetzacoalt91) [inline README.md]:
```suggestion
The update process can be launched with:
```

File: classes/Commands/CheckRequirementsCommand.php [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            $this->upgradeContainer->getTranslator(),
+            new DistributionApiService(),
+            $this->upgradeContainer->getChecksumCompare(),
+            _PS_ROOT_DIR_,
+            _PS_ADMIN_DIR_,
+            $moduleConfigPath,
+            $this->upgradeContainer->getState()->getOriginVersion()
+        );
+
+        $output->writeln('Result of prerequisite checks :');
```
Comment (Quetzacoalt91) [inline classes/Commands/CheckRequirementsCommand.php]:
```suggestion
        $output->writeln('Result of prerequisite checks:');
```

**Review by Quetzacoalt91 — APPROVED** [review]
Likely need a rebase to make the CI pass, otherwise we're good to go.

**Review by AureRita — APPROVED** [review]
Hi @M0rgan01 

Thank you for your PR, I tested it and it seems to works as you can see : 

[recording(296).webm](https://github.com/user-attachments/assets/ac7977fa-86a7-4ce1-998b-b2680205b319)


image


Tested from : 
1.7.8.9 to 8.1.7 
8.0.5 to 8.1.7 
8.1.5 to 8.1.7 
8.1.7 to 9.0.0
8.1.5 to 9.0.0
8.0.5 to 9.0.0

Because the PR seems to works as expected, It's QA :heavy_check_mark: 

Thank you

**Pull request merged**